service/rpc/share.go

```diff
@@ -45,6 +47,12 @@ func (h *Handler) handleSharesByNamespaceRequest(w http.ResponseWriter, r *http.
 		writeError(w, http.StatusBadRequest, namespacedSharesEndpoint, err)
 		return
 	}
+	// sanity check nID size
+	if len(nID) != ipld.NamespaceSize {
```

Comment:
This should be done inside our code/service or even component instead so that users who directly access the service won't get the same error. Pls also make it return the error via `fmt.Errorf()` so that you can write the error msg as a response.

Reply:
In general, aim to have such checks on the general code path.

Reply:
https://github.com/celestiaorg/celestia-node/pull/723/commits/377bb08cea2feb2ad02abb7b97f4d46321024a34